### PR TITLE
switched to pyproject.toml, update rq and dependencies

### PR DESCRIPTION
**Pull request by b97pla**
**What problems does this PR solve?**
The versions specified for the dependencies were quite outdated and when deployed with pipenv on Miarka, the resolved combination of versions was not functional. This updates the dependencies and also changes deployment to use `pip`/`setuptools` and a `pyproject.toml` file. 

**How has the changes been tested?**
Deployed locally, as well as deployed in a staging context on Miarka.

**Reasons for careful code review**
These changes break the deployment in `system-management`

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+name = "archive-verify"
+version = "1.1.0"
+authors = [
+    {name = "SNP&SEQ Technology Platform, Uppsala University", email = "seq@medsci.uu.se" },
+]
+keywords = ["bioinformatics"]
+description = "Micro-service for downloading and verifying archives with DSMC"
+readme = "README.md"
+license = {file = "LICENSE"}
+requires-python = ">=3.7"
```
Comment (slohse) [inline pyproject.toml]: This is inconsistent with what it says in the README (there it says 3.9).

**Review by slohse — APPROVED** [review]
Good work, I had just one tiny nitpick. :+1:

**Comment by slohse** [comment]
Do we have a plan regarding fixing the deployment via `system-management`?

File: pyproject.toml (hunk 1 above)
Comment (b97pla) [reply pyproject.toml]: fair enough, at some point I felt a bit conflicted about what to base the requirement on (other than what my current dev environment uses) and I think 3.7 was what I found among dependencies... anyways, thanks for pointing out, I'll fix the inconsistency

**Comment by b97pla** [comment]
I haven't created any ticket or anything, perhaps @matrulda has thoughts on this.. I'd be happy to at least suggest fixes (don't think I have proper access or knowledge to actually test or validate)

**Comment by matrulda** [comment]
@slohse Could you create a ticket about updating Sysman? We will in reality probably take care of it when validating this version, but it's good to document that it needs to be done.

**Comment by slohse** [comment]
:+1: https://snpseq.atlassian.net/browse/DEVELOP-2471

**Pull request merged**